src/core/action-creator.ts

```diff
@@ -41,38 +42,140 @@ export function createActions(flux: FluxCapacitor) {
         action(Actions.FETCH_RECOMMENDATIONS_PRODUCTS, null, metadata),
 
       // request action creators
-      updateSearch: (search: Actions.Payload.Search): Actions.UpdateSearch =>
-        action(Actions.UPDATE_SEARCH,
-          'query' in search ? { ...search, query: search.query && search.query.trim() } : search,
-          {
-            ...metadata,
-            validator: {
-              payload: [{
-                func: ({ query }) => !('query' in search) || !!query || query === null,
-                msg: 'search term is empty'
-              }, {
-                func: ({ query }, state) => query !== Selectors.query(state) || query === null,
-                msg: 'search term is not different'
-              }]
-            }
-          }),
+      updateSearch: (search: Actions.Payload.Search): Actions.UpdateSearch => {
+        const searchActions: any = [actions.resetPage()];
+
+        if ('query' in search) {
+          searchActions.push(...actions.updateQuery(search.query));
+        }
+        if ('clear' in search) {
+          searchActions.push(...actions.checkAndResetRefinements(search));
+        }
+        if ('navigationId' in search) {
+          if ('index' in search) {
+            searchActions.push(...actions.selectRefinement(search.navigationId, search.index));
+          } else if (search.range) {
+            searchActions.push(...actions.addRefinement(search.navigationId, search.low, search.high));
+          } else if ('value' in search) {
+            searchActions.push(...actions.addRefinement(search.navigationId, search.value));
+          }
+        }
+
+        return searchActions;
+      },
 
-      addRefinement: (field: string, valueOrLow: any, high: any = null) =>
-        actions.updateSearch(refinementPayload(field, valueOrLow, high)),
+      checkAndResetRefinements: ({ low, high, value, navigationId, range, clear }: Actions.Payload.Search):
+        Actions.CheckAndResetRefinements => {
```

Comment:
rather than make this an action creator, why not make it a discriminator? like `shouldResetRefinements()` which just returns a `boolean`. That way you can just add it to the `if` check for adding `resetRefinements()` to the `updateSearch()` batch action